plugins/modules/volume_cloudinit.py

```diff
@@ -130,7 +130,20 @@ def cloud_localds(volume_format,
     #       because it is not available on Red Hat Enterprise Linux 8 / 9 and CentOS 8 / 9.
     #       Ref.: https://salsa.debian.org/cloud-team/cloud-utils/-/blob/master/bin/cloud-localds
 
-    cmd = 'cloud-localds'
+    cloud_localds_path =  module.run_command('which cloud-localds')
+    if cloud_localds_path:
+        cmd = cloud_localds_path
+    else:
+        alt_path = 'if [ -d "/usr/local/cloud-localds" ]; then  echo "true" fi'
+        if alt_path == 'true':
+            cmd = '/usr/local/cloud-localds'
+        else:
+            find_output = module.run_command('find / | grep cloud-localds')
+            if find_output:
+                cmd = find_output.strip()
+            else:
+                return module.fail_json(msg="could not find cloud-localds your libvirt-host machine!")
+
```

Comment:
i have removed the kvm_host_os variable and replaced it with a little path check

Reply:
Please simply copy cloud-localds to a path where the system can find it, such as `/usr/local/bin` or `/usr/local/sbin` depending on your OS. There is no need to implement code for that, so please drop the changes to this file.